File: Cmdlets/GetCultureAndRegionInfoCommand.vb
Imports System.Management.Automation

<Cmdlet("Get", "CultureAndRegionInfo")>
Public Class GetCultureAndRegionInfoCommand
	Inherits CultureAndRegionInfoCommandBase

	<Parameter()>
	Public Property Name() As String

	Protected Overrides Sub BeginProcessing()

		If String.IsNullOrEmpty(Name) Then
			WriteObject(CultureAndRegionInfoDictionary, True)
		Else
			WriteObject(CultureAndRegionInfoDictionary(Name))
		End If
	End Sub
End Class

File: C#/Cmdlets/Cmdlets/GetCultureAndRegionInfoCommand.cs
using System.Management.Automation;

namespace TIKSN.Cmdlets
{
    [Cmdlet("Get", "CultureAndRegionInfo")]
    public class GetCultureAndRegionInfoCommand : CultureAndRegionInfoCommandBase
    {
        [Parameter()]
        public string Name { get; set; }

        protected override void BeginProcessing()
        {
            if (string.IsNullOrEmpty(Name))
            {
                WriteObject(CultureAndRegionInfoDictionary, true);
            }
            else
            {
                WriteObject(CultureAndRegionInfoDictionary[Name]);
            }
        }
    }
}